lib/brainstem/concerns/controller_dsl.rb

```diff
@@ -0,0 +1,306 @@
+require 'brainstem/concerns/inheritable_configuration'
+require 'active_support/core_ext/object/with_options'
+
+# brainstem_params do
+#   actions :index, :show do
+#     presents :project
+#   end
+#
+#   actions :edit do
+#     model_params do |param|
+#       # doc for params[:project][:sprocket_name]
+#       param.valid :sprocket_name,
+#         info: "error message"
+#
+#       # doc for params[:project][:sub_stories]
+#       param.valid :sub_stories,
+#         info: "is recursive and an array",
+#         recursive: true
+#     end
+#
+#     # doc for params[:only_one]
+#     valid :only_one,
+#       info: "only one"
+#   end
+#
+#   transform :human_readable_sprocket_attr => :some_bizarre_database_field
+#
+#   actions :create do
+#     valid :sprocket_project_id,
+#       info: "error message"
+#   end
+#
+#   actions :show do
+#     presents # by default, :sprocket for SprocketsController
+#   end
+# end
+
+
+module Brainstem
+  module Concerns
+    module ControllerDSL
+      extend ActiveSupport::Concern
+      include Brainstem::Concerns::InheritableConfiguration
+
+      DEFAULT_BRAINSTEM_PARAMS_CONTEXT = :_default
+
+      included do
```

Comment:
We will need to call `clear_configuration!` on controllers that include this, I think, to support Rails class reloading.  Should be added to `Brainstem.reset!`